src/ray/core_worker/transport/direct_task_transport.cc

```diff
@@ -607,13 +607,28 @@ void CoreWorkerDirectTaskSubmitter::RequestNewWorkerIfNeeded(
           RequestNewWorkerIfNeeded(scheduling_key);
 
         } else {
-          if (IsRayletFailed(RayConfig::instance().RAYLET_PID())) {
-            RAY_LOG(WARNING)
-                << "The worker failed to receive a response from the local "
-                   "raylet because the raylet is crashed. Terminating the worker.";
-            QuickExit();
+          if (status.IsIOError()) {
```

Comment:
Is it correct to treat all IO errors as raylet unavailable? Why not only consider the specific unavailable error as before?

Reply:
Oh, yea. Added it back.